Support multiline GitHub App keys in automation env

Handoff Source: Founder triage automation on 2026-04-16; reviewed current origin/main@a299f60832c361eb3fce6c9660782ac890fdae64 after merged PR #5998, ROADMAP.md, docs/FEATURE_DISCOVERY.md, docs/status/FEATURE_DISCOVERY.md, docs/status/NEXT_STEPS_CANONICAL.md, docs/status/ACTIVE_EXECUTION_ISSUES.md, recent founder-triage/founder-review/Engineering Automation 2/Aragora Overnight Steward memories, open PRs/issues, recent merged PRs, proof publication branches, remote codex/factory/boss-harvest branches, focused GitHub App auth code/docs/tests, and a fallback-only Aragora spec run.

Priority: HIGH

Task Title: Support multiline GitHub App keys in automation env

Why Now: PR #5998 just moved proof-first queue reconciliation and boss GitHub calls toward GitHub App installation tokens, but the new `.env.automation` parser only reads one physical line for `GITHUB_APP_PRIVATE_KEY` while the setup guide documents a quoted multiline PEM. Operators following the documented setup will silently fall back to the user `gh` token when strict mode is off, preserving the GraphQL/rate-limit/publisher failures #5998 was meant to avoid. This directly supports the current proof-first gate and likely sits upstream of fresh benchmark/publication branches that still have no draft PR.

Repo Evidence: On `origin/main@a299f60832c361eb3fce6c9660782ac890fdae64`, `aragora/swarm/github_app_auth.py:37-50` parses `.env.automation` line-by-line, splits each physical line on `=`, and strips quotes from that single line. `docs/guides/GITHUB_APP_SETUP.md:66-69` documents `GITHUB_APP_PRIVATE_KEY` as a quoted multiline PEM. A local reproducer wrote `GITHUB_APP_ID`, `GITHUB_APP_INSTALLATION_ID`, and a quoted multiline `GITHUB_APP_PRIVATE_KEY` into a temp `.env.automation`; `load_github_app_config({'ARAGORA_AUTOMATION_ENV_FILE': path})` returned only `'-----BEGIN RSA PRIVATE KEY-----'`. `tests/swarm/test_github_app_auth.py` currently covers only `*_PRIVATE_KEY_PATH` and unconfigured fallback, not inline multiline or escaped-newline private keys. Founder-review independently reproduced that `jwt.encode(..., algorithm='RS256')` fails with `InvalidKeyError` for the truncated inline key while the same generated RSA key signs successfully through `GITHUB_APP_PRIVATE_KEY_PATH`. Open PR list is empty; open boss-ready issue #6000 targets benchmark corpus freshness, not auth; targeted issue searches for multiline GitHub App private keys and `.env.automation` returned no open issue; targeted PR search only found merged PR #5998; remote branches touching GitHub App/proof auth are either merged #5998 or older publisher workflow branches, not this parser bug. Attempted `gh issue create` for this novel task failed with `error connecting to api.github.com`; publisher should retry issue creation with label `boss-ready`.

Acceptance Criteria: `load_github_app_config()` supports documented quoted multiline PEM values and common escaped-newline values for `GITHUB_APP_PRIVATE_KEY` and `ARAGORA_GITHUB_APP_KEY`. Preserve existing `GITHUB_APP_PRIVATE_KEY_PATH` / `GH_APP_PRIVATE_KEY_PATH` / `ARAGORA_GITHUB_APP_PRIVATE_KEY_PATH` behavior and precedence. Add focused tests that fail on current main for documented multiline `.env.automation` content and escaped-newline inline key content, ideally using a generated or fixture RSA key to prove `jwt.encode(..., algorithm='RS256')` can sign from inline config. Keep missing/malformed App config fallback behavior explicit and covered, including strict-mode behavior.

Validation: Run `python3 -m pytest tests/swarm/test_github_app_auth.py tests/scripts/test_reconcile_proof_first_queue.py -q`. Run a small local JWT-signing regression using an RSA key loaded from inline env content. Re-run `python3 scripts/reconcile_proof_first_queue.py --json` with no App config to confirm graceful fallback remains. If publishing a PR, run `bash scripts/automation_pr_preflight.sh origin/main HEAD` and confirm no benchmark publisher or playground receipt files are mixed into the diff.

Expiration Hours: 72

Backup Task: If this GitHub App key parser task is already fixed or actively claimed before pickup, validate and publish `origin/codex/codex-20260416-164000-playground-receipts` as the next task: it is based on current `origin/main`, has no PR, changes only `aragora/server/handlers/playground.py` and `tests/handlers/test_playground.py`, and fixes the receiptless public playground debate regression. If that branch is also published/merged, take open boss-ready issue #6000 for benchmark corpus freshness next; branch `origin/codex/codex-20260416-161927-a2efcdff` already contains a one-file test implementation but also has no PR.

## 2026-04-16T16:23:49Z

- Runtime: about `1:33:31` since the last provided founder-triage run timestamp (`2026-04-16T14:50:18Z`), and about `1:29:30` after the freshest existing founder-triage memory entry.
- Read `ROADMAP.md`, `docs/FEATURE_DISCOVERY.md`, canonical `docs/status/FEATURE_DISCOVERY.md`, `docs/status/NEXT_STEPS_CANONICAL.md`, `docs/status/ACTIVE_EXECUTION_ISSUES.md`, current founder-triage memory, and recent `founder-review`, `engineering-automation-2`, and `aragora-overnight-steward` memories.
- Current `origin/main` / `HEAD` is `a299f6083` (`fix(autonomy): use GitHub App token for proof queue (#5998)`). Recent merged PRs reviewed include `#5998`, `#5996`, `#5990`, `#5987`, `#5985`, `#5981`, `#5980`, `#5979`, `#5978`, `#5977`, `#5976`, `#5971`, `#5965`, `#5964`, and `#5963`.
- Live GitHub state: `gh auth status` succeeds for `an0mium`; `gh pr list --state open` returned `[]`; open `boss-ready` issues are `#5962` (`[from #5894] Create benchmark corpus freshness test`) and `#5893` (`[TW-02] Add benchmark truth surface freshness validation tests`). `#5893` is stale on current main because `tests/benchmarks/test_truth_surface_consistency.py` exists and passes (`2 passed in 0.31s`); `#5962` remains a valid benchmark backup because `tests/benchmarks/test_corpus_freshness.py` is absent.
- Current proof surfaces in repo remain the April 15 status docs: B0 last updated `2026-04-15T23:05:41Z`, complete `5/5`, truth/no-rescue/merged-only rates `100%`; TW03 last updated `2026-04-15T23:05:42Z`, repeated rescue classes `0`. A fresh publication branch exists at `origin/benchmark-truth-publication/24517976139` (`5754ed0f1 docs(status): refresh recurring trust-loop surfaces`, generated around `2026-04-16T15:09Z`) and has no PR (`gh pr list --head benchmark-truth-publication/24517976139 --state all` returned `[]`), but #5998 just landed the GitHub App token fix for the proof queue/publisher path.
- Current contract checks: `python3 scripts/validate_openapi_routes.py --spec docs/api/openapi.json --json` reports `96.1%` coverage, `89` missing handler routes, and `39` orphaned spec routes. `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json` passes within budget with `18` missing-from-both routes. `python3 scripts/verify_sdk_contracts.py --strict --baseline scripts/baselines/verify_sdk_contracts.json` passes with no baseline regressions.
- Reproduced the previous playground receipt regression on current `main`: `pytest -q tests/handlers/test_playground.py::TestMockDebate::test_response_has_receipt` failed after `137.28s` because the successful `/api/v1/playground/debate` response body lacked `receipt`.
- Direct code evidence on current main:
  - `tests/handlers/test_playground.py:407-415` expects the playground debate response to include `receipt.receipt_id`, `receipt.signature`, and `signature_algorithm == "SHA-256-content-hash"`.
  - `aragora/server/handlers/playground.py:3428-3446` accepts any `_run_live_debate()` result with status below 400 and returns it through `_persist_and_respond()`.
  - `aragora/server/handlers/playground.py:3615-3641` can construct a package-backed successful response with `receipt: None` and `receipt_hash: None` when `result.receipt` is absent.
- Duplication checks for the playground receipt task:
  - No open PRs.
  - `gh issue list` / `gh search` for `playground receipt` and `mock debate receipt` returned `[]` when GitHub API was reachable; some broader `gh search` calls transiently failed with `error connecting to api.github.com`.
  - Remote branch `origin/codex/codex-20260416-150230-71a60779` exists with commit `39fcb45ae fix(playground): preserve mock debate receipts`. The commit itself changes only `aragora/server/handlers/playground.py` and `tests/handlers/test_playground.py` (`112 insertions`, `18 deletions`) and adds receipt-preserving fallback tests.
  - The branch is behind/diverged from `origin/main` because its merge base is `54f0ccf1b` (`#5963`) and current `main` has `a299f6083` (`#5998`). A direct `origin/main..branch` diff would appear to remove the newly merged #5998 GitHub App token files, so the worker should replay or rebase only the playground receipt commit rather than publish the stale branch tip as-is.
- Aragora usage: `aragora spec` was run for the post-#5998 prioritization decision and returned fallback-only output because the `spec-agent` circuit breaker is open.
- Decision: primary recommendation is not a new issue; it is a bounded branch-rescue/publish task for the existing playground receipt fix. No GitHub issue was opened because the work is not novel: a fresh remote codex branch already contains the scoped implementation, but it needs rebasing/replay onto current `main` without regressing #5998.
- Current run time: `2026-04-16T16:23:49Z`.

Handoff Source: Founder triage automation on 2026-04-16; reviewed origin/main@a299f6083 after merged PRs #5963, #5964, #5965, #5971, #5976-#5981, #5985, #5987, #5990, #5996, and #5998; ROADMAP.md, docs/FEATURE_DISCOVERY.md, docs/status/FEATURE_DISCOVERY.md, docs/status/NEXT_STEPS_CANONICAL.md, docs/status/ACTIVE_EXECUTION_ISSUES.md, current and recent automation memories, open PRs/issues, proof/status surfaces, route/SDK validators, remote codex/factory/boss-harvest/benchmark branches, focused playground receipt repro, and a fallback-only Aragora spec run.
Priority: HIGH
Task Title: Rebase playground receipt fix onto current main
Why Now: Current `main` still has a demo-readiness and receipt-correctness regression: the public `/api/v1/playground/debate` path can return a successful debate payload without the receipt artifact required by the existing trust contract. This is stronger than opening a new broad task because a fresh remote codex branch already contains a narrow two-file fix, but that branch is based on `#5963` and would conflict semantically with the newly merged `#5998` GitHub App token work if published as-is. The highest-leverage coding run is to replay/rebase only the playground receipt fix onto `a299f6083` and publish it cleanly.
Repo Evidence: `pytest -q tests/handlers/test_playground.py::TestMockDebate::test_response_has_receipt` fails on `origin/main@a299f6083` after `137.28s` with `AssertionError: assert 'receipt' in {...}`. `tests/handlers/test_playground.py:407-415` expects `receipt.receipt_id`, `receipt.signature`, and `signature_algorithm == "SHA-256-content-hash"`. `aragora/server/handlers/playground.py:3428-3446` accepts any successful `_run_live_debate()` response before falling back to `_run_inline_mock_debate()`, while `aragora/server/handlers/playground.py:3615-3641` can build a successful package-backed response with `receipt: None` when `result.receipt` is absent. Remote branch `origin/codex/codex-20260416-150230-71a60779` at `39fcb45ae` already implements `fix(playground): preserve mock debate receipts`, changing only `aragora/server/handlers/playground.py` and `tests/handlers/test_playground.py` in its own commit. The branch is not merged, has no PR, and is not an ancestor of current main; `origin/main..branch` also shows stale differences in `aragora/swarm/github_app_auth.py`, `scripts/reconcile_proof_first_queue.py`, and related #5998 files because the branch predates `a299f6083`. Open PR list is empty; targeted GitHub issue/PR searches found no playground receipt issue; open boss-ready issues are #5962 and stale #5893, neither duplicates this branch-rescue task.
Acceptance Criteria: Apply the intent of `39fcb45ae` onto current `origin/main@a299f6083` without reverting or deleting the #5998 GitHub App token/proof-queue changes. `/api/v1/playground/debate` must return a receipt-shaped artifact for deterministic public mock/fallback responses, and receiptless live-package success must fail closed into the receipt-bearing mock fallback or otherwise avoid returning a trustworthy-looking receiptless success. Preserve explicit fallback labeling such as `mock_fallback` / `is_live: false`, keep scope to playground receipt correctness plus focused tests, and do not broaden into benchmark publisher or SDK parity work.
Validation: Run `pytest -q tests/handlers/test_playground.py::TestMockDebate::test_response_has_receipt` and the new/focused playground tests added by the replayed fix. Run `pytest -q tests/handlers/test_playground.py -k "receipt or mock_endpoint or receiptless_live_success"`. Run `git diff --name-only origin/main...HEAD` and confirm only playground handler/tests plus any intentional metadata are changed, not the #5998 proof-queue/auth files. If opening a PR, run `bash scripts/automation_pr_preflight.sh origin/main HEAD` and confirm required checks start from the rebased branch.
Expiration Hours: 24
Backup Task: If the playground receipt fix is already merged before pickup, take open boss-ready issue #5962 (`[from #5894] Create benchmark corpus freshness test`) next. It remains novel on current main because `tests/benchmarks/test_corpus_freshness.py` is absent, while #5893 is stale because `tests/benchmarks/test_truth_surface_consistency.py` already exists and passes. Also verify the fresh benchmark publication branch `origin/benchmark-truth-publication/24517976139` after #5998; if the publisher still fails to open a PR, inspect the auto-pr-publisher workflow before adding new benchmark breadth.

## 2026-04-16T14:54:19Z

- Runtime: about `1:04:01` since the last founder-triage run timestamp.
- Read `ROADMAP.md`, `docs/FEATURE_DISCOVERY.md`, canonical `docs/status/FEATURE_DISCOVERY.md` entrypoint context through `docs/status/NEXT_STEPS_CANONICAL.md`, `docs/status/ACTIVE_EXECUTION_ISSUES.md`, current founder-triage memory, and recent `founder-review`, `engineering-automation-2`, and `aragora-overnight-steward` memories.
- Current `origin/main` / `HEAD` is `c30dd5c1f` (`fix(automation): publish freshest handoff memory (#5961)`). Recent merged PRs reviewed include `#5954`, `#5958`, `#5959`, and `#5961`.
- Open PR duplication state before the final GitHub write degraded: `gh pr list --state open` returned `#5963` (`fix(sdk): restore prompt engine contracts`) and `#5996` (`fix(autonomy): reset recovery budgets per proof-first shift`). `#5963` covers issue `#5960` / the previous PromptEngineHandler handoff; required checks had passed on the earlier completed run, with a newer rerun still queued at the time of inspection. `#5996` was active proof-first recovery-budget work with required TypeScript SDK check still pending.
- Open boss-ready issue state before final GitHub write degraded: `#5960` (`Restore PromptEngineHandler OpenAPI and SDK contract`), `#5962` (`[from #5894] Create benchmark corpus freshness test`), and `#5897` (`[CS-01] Reconcile COMMERCIAL_OVERVIEW benchmark claims against current truth surfaces`). None duplicate the playground receipt regression.
- The prior PromptEngineHandler founder-triage recommendation is now covered by `#5960` and active PR `#5963`, so it should not be reissued as the primary task.
- Current proof surfaces remain fresh enough for this run: `docs/status/B0_BENCHMARK_TRUTH_STATUS.md` last updated `2026-04-15T23:05:41Z`, complete `5/5`, truth/no-rescue/merged-only rates `100%`; `docs/status/TW03_RESCUE_PRODUCTIZATION_STATUS.md` last updated `2026-04-15T23:05:42Z`, repeated rescue classes `0`.
- Current route/SDK truth on main:
  - `python3 scripts/validate_openapi_routes.py --spec docs/api/openapi.json --json` reports `96.1%` route coverage, `89` missing handler routes, and `39` orphaned spec routes.
  - `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json` passes within budget with `26` missing-from-both routes; `PromptEngineHandler` still appears with 8 missing routes on main, but the active PR owns that lane.
- New highest-leverage unassigned task comes from founder-review and was reproduced directly on current main: `pytest -q tests/handlers/test_playground.py::TestMockDebate::test_response_has_receipt` failed in isolation after `56.03s` because the successful `/api/v1/playground/debate` response body lacked `receipt`.
- Direct code evidence:
  - `tests/handlers/test_playground.py:407-415` expects the playground debate response to include `receipt.receipt_id`, `receipt.signature`, and `signature_algorithm == "SHA-256-content-hash"`.
  - `aragora/server/handlers/playground.py:2924-2928` defaults empty public debate requests to `source="landing"`.
  - `aragora/server/handlers/playground.py:3428-3446` accepts any `_run_live_debate()` result with status below 400 before falling back to inline mock behavior.
  - `aragora/server/handlers/playground.py:3615-3641` can build a successful package-backed debate response with `receipt: None` and `receipt_hash: None` when `result.receipt` is absent.
  - `aragora/server/handlers/playground.py:3497-3534` normalizes/persists the response and returns it, preserving the receiptless success shape.
- Duplication checks for the new task:
  - Targeted open issue searches for `playground receipt`, `playground mock debate`, and `TestMockDebate response_has_receipt` returned `[]`.
  - Targeted open PR search for the same terms returned `[]`.
  - Remote branch scan showed old/stale playground or receipt branches, but no active recent branch touching `aragora/server/handlers/playground.py` or `tests/handlers/test_playground.py` for this exact regression.
  - Open PR `#5963` targets PromptEngine contracts; open PR `#5996` targets proof-first recovery budgets; open boss-ready issues target PromptEngine, benchmark corpus freshness, and commercial proof-drift docs.
- Aragora usage: `aragora spec` was run for the playground receipt recommendation and returned fallback-only output because the `spec-agent` circuit breaker is open.
- Tried to open GitHub issue with title `Restore playground mock debate receipts` and label `boss-ready`; `gh issue create --repo synaptent/aragora --label boss-ready --body-file -` failed with exact blocker: `error connecting to api.github.com` then `check your internet connection or https://githubstatus.com`.
- Publisher retry should create one `boss-ready` issue with the handoff below; do not create a duplicate if a later run or publisher has already opened it.
- Current run time: `2026-04-16T14:54:19Z`.

Handoff Source: Founder triage automation on 2026-04-16; reviewed origin/main@c30dd5c1f7d401ef40362685d0b4b7bcf30a243a after merged PRs #5954, #5958, #5959, and #5961; ROADMAP.md, docs/FEATURE_DISCOVERY.md, docs/status/NEXT_STEPS_CANONICAL.md, docs/status/ACTIVE_EXECUTION_ISSUES.md, B0/TW03 proof surfaces, founder-triage memory, founder-review memory, Engineering Automation 2 memory, Aragora Overnight Steward memory, open PRs #5963/#5996, open boss-ready issues #5960/#5962/#5897, current OpenAPI/SDK validators, targeted GitHub searches, remote branch scans, a focused playground receipt repro, and a fallback-only Aragora spec run.
Priority: HIGH
Task Title: Restore playground mock debate receipts
Why Now: This is the strongest unassigned Track 1-4 task because it is both demo readiness and receipt correctness: the public `/api/v1/playground/debate` path can return a successful debate payload without the receipt-shaped trust artifact that the existing test requires. The current benchmark and rescue proof surfaces remain fresh enough, the PromptEngineHandler contract gap is already covered by issue #5960 and active PR #5963, and proof-first recovery budget work is already active in PR #5996. This receipt regression is novel, reproducible on current main, and small enough for one bounded coding run.
Repo Evidence: `tests/handlers/test_playground.py:407-415` expects `POST /api/v1/playground/debate` to include `receipt.receipt_id`, `receipt.signature`, and `signature_algorithm == "SHA-256-content-hash"`. On current `origin/main`, `pytest -q tests/handlers/test_playground.py::TestMockDebate::test_response_has_receipt` fails in isolation after 56.03s with `AssertionError: assert 'receipt' in {...}`. `PlaygroundHandler.handle_post()` defaults the source to `landing` for empty public debate bodies at `aragora/server/handlers/playground.py:2924-2928`. `_run_debate()` accepts any `_run_live_debate()` result with status < 400 at `aragora/server/handlers/playground.py:3428-3446` before falling back to the inline mock at `:3459-3471`. `_run_debate_with_package()` sets `receipt_data = None` when package debate results lack `result.receipt` at `aragora/server/handlers/playground.py:3615-3619`, then returns a response with `receipt: None` / `receipt_hash: None` at `:3621-3641`; `_persist_and_respond()` normalizes and persists the response at `:3497-3534`. Founder-review also saw the broader integration-smoke command fail after 1,405 passing tests at the same assertion. Duplication checks found no open issue or PR for `playground receipt`, `playground mock debate`, or `TestMockDebate response_has_receipt`; recent branch scans show only old/stale playground or receipt branches, not an active branch touching `aragora/server/handlers/playground.py` or `tests/handlers/test_playground.py` for this exact regression.
Acceptance Criteria: `/api/v1/playground/debate` always returns a receipt-shaped trust artifact for public mock/fallback debate responses, including the package-backed live-debate-success path where `result.receipt` is absent; alternatively, classify that path as unavailable and fall back to `_run_inline_mock_debate()` so the existing receipt contract is preserved. `tests/handlers/test_playground.py::TestMockDebate::test_response_has_receipt` passes in isolation without network/API keys. The fix preserves truthful fallback labeling such as `mock_fallback` when live agents are unavailable and does not make a receiptless live-package result look more trustworthy than it is. Keep scope limited to playground debate receipt correctness and focused tests.
Validation: Run `pytest -q tests/handlers/test_playground.py::TestMockDebate::test_response_has_receipt`. Run the focused playground handler suite, or if runtime allows `pytest tests/handlers/ tests/debate/test_arena*.py tests/memory/ -x --timeout=60 -q --ignore=tests/connectors -k "not benchmark and not slow"` to confirm the integration-smoke assertion is gone. Re-run targeted duplicate checks before implementation: open PRs currently include #5963 and #5996 but neither targets playground receipts; open boss-ready issues #5960, #5962, and #5897 do not duplicate this task. Aragora spec was attempted for this recommendation and returned fallback-only output because the spec-agent circuit breaker is open.
Expiration Hours: 48
Backup Task: If playground receipt correctness is already fixed before pickup, take open boss-ready issue #5962 (`[from #5894] Create benchmark corpus freshness test`) next: no open PR currently matches `benchmark corpus freshness` / `test_corpus_freshness`, and it directly supports the current proof-first gate by detecting stale truth outcomes in `docs/benchmarks/corpus.json`. If #5962 is also active or resolved, only then rescue active PR #5963 or #5996 if a concrete required check has failed.

## 2026-04-16T13:54:06Z

- Runtime: about `1:03:48` since the last founder-triage run timestamp.
- Read `ROADMAP.md`, `docs/FEATURE_DISCOVERY.md`, canonical `docs/status/FEATURE_DISCOVERY.md`, `docs/status/NEXT_STEPS_CANONICAL.md`, `docs/status/ACTIVE_EXECUTION_ISSUES.md`, current founder-triage memory, and recent `founder-review`, `engineering-automation-2`, and `aragora-overnight-steward` memories.
- Current local `origin/main` / `HEAD` is `2666755b0` (`[codex] Restore task queue OpenAPI and SDK contract (#5958)`). The prior founder-triage TaskQueueHandler recommendation is now closed by issue `#5953` / PR `#5958`; recent merged PRs also include `#5954`, `#5952`, `#5951`, and `#5950`.
- Open PR duplication state: `gh pr list --state open` returned one active PR, `#5959` (`[codex] Route decision analytics modular dispatch`) on `codex/codex-20260416-132304-ca2b476c`, carrying `codex` and `codex-automation`. It covers the latest founder-review decision-analytics modular-dispatch bug, so no duplicate was recommended.
- Open boss-ready issue state: `gh issue list --state open --label boss-ready` returned `[]`. `gh issue view 5953` showed `Restore TaskQueueHandler OpenAPI and SDK contract` is closed at `2026-04-16T13:36:26Z`.
- Proof surfaces remain fresh enough for this run: `docs/status/B0_BENCHMARK_TRUTH_STATUS.md` last updated `2026-04-15T23:05:41Z`, complete `5/5`, with `100%` truth/no-rescue/merged-only rates; `docs/status/TW03_RESCUE_PRODUCTIZATION_STATUS.md` last updated `2026-04-15T23:05:42Z` with zero repeated rescue classes.
- Current route/SDK truth on main:
  - `python3 scripts/validate_openapi_routes.py --spec docs/api/openapi.json --json` reports `96.1%` route coverage, `89` missing handler routes, and `39` orphaned spec routes.
  - `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json` passes within budget with `26` missing-from-both routes; `PromptEngineHandler` accounts for `8` of them (`py: -8`, `ts: -8`).
  - `python3 scripts/verify_sdk_contracts.py --strict --baseline scripts/baselines/verify_sdk_contracts.json` passes with no new baseline regressions.
- Direct PromptEngineHandler evidence:
  - `PromptEngineHandler` is already present in `ALL_HANDLERS`, so this is not a lazy-discovery gap.
  - `aragora/server/handlers/prompt_engine/handler.py` documents GET `/api/prompt-engine/runs`, GET `/api/prompt-engine/runs/{id}`, and POST `/run`, `/decompose`, `/interrogate`, `/research`, `/specify`, `/validate`.
  - `can_handle()` allows GET only for run list/detail and POST for action endpoints; `handle()` dispatches GET to `_handle_get`; `handle_post()` / `handle_POST()` dispatch action endpoints.
  - `pytest -q tests/server/handlers/test_prompt_engine_handler.py` passed (`26 passed, 7 warnings`).
  - Fresh `python3 scripts/export_openapi.py --output-dir /tmp/founder-triage-prompt-engine-openapi` shows both `/api/prompt-engine/*` and `/api/v1/prompt-engine/*` prompt-engine routes as autogenerated placeholders; critically, `/runs` and `/runs/{run_id}` are exported as `post` only despite the handler exposing GET reads.
  - Existing generated TypeScript OpenAPI types mirror those placeholder POST operations; no dedicated Python/TypeScript prompt-engine SDK namespace coverage was found beyond generated type definitions.
- Duplication checks:
  - Targeted open PR search initially degraded because of an invalid field request, then broader `gh search prs/issues` for prompt-engine failed with `error connecting to api.github.com`; earlier `gh pr list` and `gh issue list --label boss-ready` worked.
  - `gh issue list --search 'prompt-engine OR PromptEngineHandler OR prompt engine SDK OR prompt-to-spec SDK'` returned only unrelated issue `#2976` before later GitHub search degraded.
  - Remote branch scan over recent `origin/codex`, `origin/factory`, and `origin/aragora/boss-harvest` refs found no recent prompt-engine contract branch; the only active open PR branch is `origin/codex/codex-20260416-132304-ca2b476c` for decision analytics.
- Aragora usage: `aragora spec` was run for the PromptEngineHandler contract task, but returned fallback-only output because the spec-agent circuit breaker is open.
- Decision: recommend a novel bounded task to restore PromptEngineHandler OpenAPI and SDK contract coverage. This is lower priority than the already-active `#5959` decision-analytics PR, but it is the next concrete live product/integration correctness gap after TaskQueueHandler and agent-evolution fixes landed.
- Tried to open GitHub issue with title `Restore PromptEngineHandler OpenAPI and SDK contract` and label `boss-ready`; `gh issue create --repo synaptent/aragora --label boss-ready --body-file -` failed with exact blocker: `error connecting to api.github.com` then `check your internet connection or https://githubstatus.com`.
- Publisher retry should create one `boss-ready` issue with the handoff below; do not create a duplicate if a later run or publisher has already opened it.
- Current run time: `2026-04-16T13:54:06Z`.

Handoff Source: Founder triage automation on 2026-04-16; reviewed current main at 2666755b0 / PR #5958, roadmap/status docs, B0/TW03 proof surfaces, recent founder-review / Engineering Automation 2 / Aragora Overnight Steward memories, open PR #5959, issue #5953 closure, current route and SDK validators, prompt-engine handler/runtime tests, remote codex/factory/boss-harvest branches, and a fallback-only Aragora spec run.
Priority: HIGH
Task Title: Restore PromptEngineHandler OpenAPI and SDK contract
Why Now: The prompt-to-spec / debate / execution wedge is a live Track 3 product path, but its public contract remains misleading after the higher-priority TaskQueue and agent-evolution fixes landed. `PromptEngineHandler` implements live GET reads for persisted runs and POST actions for run/decompose/interrogate/research/specify/validate, while generated OpenAPI still emits autogenerated POST-only placeholders for all eight prompt-engine paths and both SDKs expose zero PromptEngineHandler route coverage. This is now the next bounded integration-correctness task behind active PR #5959, and it prevents client generation from encoding the wrong methods for the prompt-engine read paths.
Repo Evidence: `aragora/server/handlers/prompt_engine/handler.py:3-11` documents GET `/api/prompt-engine/runs`, GET `/api/prompt-engine/runs/{id}`, and POST action endpoints. `aragora/server/handlers/prompt_engine/handler.py:55-64` declares those eight routes; `:66-77` allows GET only for `/runs` and `/runs/{id}` while POST covers action endpoints; `:79-84` dispatches GET through `_handle_get`; and `:87-115` dispatches POST action handlers. `pytest -q tests/server/handlers/test_prompt_engine_handler.py` passes (`26 passed`), including `can_handle` tests for GET run reads and POST action endpoints. Fresh `python3 scripts/export_openapi.py --output-dir /tmp/founder-triage-prompt-engine-openapi` shows `/api/prompt-engine/runs`, `/api/prompt-engine/runs/{run_id}`, and their `/api/v1/*` aliases as `post` only with `x-autogenerated: true`; all other prompt-engine routes are also autogenerated placeholders. `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json` reports 26 routes missing from both SDKs, including all eight `/api/prompt-engine/*` routes, and lists `PromptEngineHandler: 8 routes (py: -8, ts: -8)`. `PromptEngineHandler` is already in `ALL_HANDLERS`, so the gap is contract metadata/SDK coverage, not handler discovery. Open PR #5959 covers decision analytics modular dispatch and should not be duplicated; issue #5953 is closed by #5958; open `boss-ready` issue list was empty; targeted prompt-engine PR/issue searches found no matching active work before GitHub search degraded; remote codex/factory/boss-harvest branch scan found no recent prompt-engine contract branch.
Acceptance Criteria: Add explicit OpenAPI definitions or route metadata for the live PromptEngineHandler contract instead of relying on autogenerated placeholders. `/api/prompt-engine/runs` and `/api/prompt-engine/runs/{run_id}` plus `/api/v1/prompt-engine/runs` and `/api/v1/prompt-engine/runs/{run_id}` must expose GET, not POST-only placeholders. The action endpoints `/run`, `/decompose`, `/interrogate`, `/research`, `/specify`, and `/validate` must remain POST and gain non-placeholder request/response metadata aligned with existing handler behavior. Add focused OpenAPI tests proving generated schema methods match `PromptEngineHandler.can_handle()` / dispatch semantics and that no prompt-engine path is left with `x-autogenerated: true`. Extend Python and TypeScript SDK coverage for the prompt-engine namespace with route-mapping tests for run list/detail plus each action endpoint. Keep the change scoped to PromptEngineHandler contract coverage and do not expand into unrelated SDK parity baseline debt.
Validation: Run `pytest -q tests/server/handlers/test_prompt_engine_handler.py`. Run the new focused OpenAPI prompt-engine contract tests. Run `python3 scripts/export_openapi.py --output-dir /tmp/prompt-engine-contract-openapi` and a direct JSON check that `/api/prompt-engine/runs` and `/api/prompt-engine/runs/{run_id}` expose GET, action endpoints expose POST, and none of the prompt-engine paths carry `x-autogenerated: true`. Run `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json` and confirm the PromptEngineHandler missing-from-both gap is gone. Run focused Python SDK prompt-engine tests and focused TypeScript SDK namespace tests, or `npm --prefix sdk/typescript run typecheck` when dependencies are available. Run `python3 scripts/verify_sdk_contracts.py --strict --baseline scripts/baselines/verify_sdk_contracts.json` to ensure no new SDK/API drift.
Expiration Hours: 48
Backup Task: If PromptEngineHandler contract coverage is already fixed before pickup, wait for #5959 to finish and repair any concrete failed required check on that decision-analytics modular-dispatch PR; otherwise target the next SDK parity wedge with live UI impact, `FeedbackHubHandler`, which still appears in `check_sdk_parity.py` with four missing SDK routes.

## 2026-04-16T12:53:06Z

- Runtime: about `1:02:49` since the last founder-triage run timestamp.
- Read `ROADMAP.md`, `docs/FEATURE_DISCOVERY.md`, canonical `docs/status/FEATURE_DISCOVERY.md`, `docs/status/NEXT_STEPS_CANONICAL.md`, `docs/status/ACTIVE_EXECUTION_ISSUES.md`, current founder-triage memory, and recent `founder-review`, `engineering-automation-2`, and `aragora-overnight-steward` memories.
- Current `origin/main` / `HEAD` is `4846187b8` (`fix(openapi): stabilize decision analytics generation (#5952)`). The prior founder-triage primary (`Fix decision analytics OpenAPI generation drift`) is now resolved on main by `#5952`; the earlier token blacklist, agent-evolution SDK, and system-intelligence SDK handoffs are also resolved by `#5950`, `#5948`, and `#5951`.
- Live GitHub duplication checks: `gh pr list --state open` returned `[]`; `gh issue list --label boss-ready` returned `[]`; targeted open issue search found existing issue `#5953` (`Restore TaskQueueHandler OpenAPI and SDK contract`) with no labels. Targeted searches also found stale issue `#5929` and broad issue `#5922`, but no open PR or branch that fixes TaskQueueHandler OpenAPI/SDK contract truth.
- Remote branch scan found the newest branch `origin/codex/codex-20260416-124216-0a47a9dc` (`fix(openapi): publish agent evolution action methods`) touching agent-evolution OpenAPI/SDK generated artifacts only; it does not cover TaskQueueHandler. `origin/codex/codex-20260416-120620-c7b6b422` is the already-merged decision-analytics generation fix. Older inbox/chat branches are stale relative to merged fixes.
- Current proof surfaces remain fresh enough for this run: `docs/status/B0_BENCHMARK_TRUTH_STATUS.md` was last updated `2026-04-15T23:05:41Z`; `docs/status/TW03_RESCUE_PRODUCTIZATION_STATUS.md` was last updated `2026-04-15T23:05:42Z`.
- Current route validator state on main: `python3 scripts/validate_openapi_routes.py --spec docs/api/openapi.json --json` reports `96.0%` coverage, `91` missing handler routes, and `39` orphaned spec routes. It still reports `/api/v1/tasks/queue/{param}` and `/api/v1/tasks/leases/{param}` in `missing_in_spec`.
- Current strict SDK parity passes only within the existing budget: `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json` reports `29` missing-from-both routes, `0` baseline regressions, and budget `29/29`. `TaskQueueHandler` has `5` routes with `0` Python and `0` TypeScript SDK coverage; `/api/tasks/queue`, `/api/tasks/leases`, and `/api/tasks/salvage` remain missing from both SDKs.
- Direct local proof for TaskQueueHandler contract drift: `docs/api/openapi.json` has `/api/v1/tasks/queue`, `/api/v1/tasks/leases`, and `/api/v1/tasks/salvage` as `post`-only `x-autogenerated` placeholders. It is missing `/api/v1/tasks/queue/{task_id}`, `/api/v1/tasks/queue/{task_id}/claim`, `/api/v1/tasks/queue/sync`, and `/api/v1/tasks/leases/{lease_id}/complete`, even though `TaskQueueHandler.handle()` dispatches GET queue/lease/salvage reads and `handle_post()` dispatches sync/claim/heartbeat/release/complete actions.
- Focused runtime baseline is green: `pytest -q tests/handlers/tasks/test_queue.py` => `7 passed in 1.39s`.
- Aragora usage: `aragora spec` was run for this candidate and degraded to fallback-only output because the `spec-agent` circuit breaker is open.
- GitHub publishing state: existing issue `#5953` already contains the full Engineering Automation 2 handoff body, so no duplicate issue was opened. Attempted `gh issue edit 5953 --add-label boss-ready`, but it failed with exact blocker: `error connecting to api.github.com` then `check your internet connection or https://githubstatus.com`. Publisher retry should add `boss-ready` to `#5953` rather than creating a duplicate issue.
- Current run time: `2026-04-16T12:53:06Z`.

Handoff Source: Founder triage automation on 2026-04-16; reviewed origin/main@4846187b8, roadmap/status docs, recent founder-review/engineering-automation-2/overnight-steward memories, live PR/issues, recent merged PRs #5948/#5950/#5951/#5952, current route/SDK validators, remote codex branches, existing issue #5953, and a fallback-only Aragora spec run.
Priority: HIGH
Task Title: Restore TaskQueueHandler OpenAPI and SDK contract
Why Now: Task queue, lease lifecycle, and salvage endpoints are reliability-substrate surfaces for bounded autonomous work, but current public artifacts misdescribe them. Runtime handler tests pass, while OpenAPI advertises bogus post-only placeholders for the base task queue paths and both SDKs expose zero TaskQueueHandler coverage. This can mislead generated clients and encode the wrong contract before the next SDK parity improvement.
Repo Evidence: `aragora/server/handlers/tasks/queue.py:35-40` declares `/api/v1/tasks/queue`, `/api/v1/tasks/queue/*`, `/api/v1/tasks/leases`, `/api/v1/tasks/leases/*`, and `/api/v1/tasks/salvage`. `aragora/server/handlers/tasks/queue.py:117-128` handles GET reads for leases, salvage, queue stats, queue list, and task detail; `:147-163` handles POST queue sync, task claim, lease heartbeat, lease release, and lease complete actions. `docs/api/openapi.json` currently lists `/api/v1/tasks/queue`, `/api/v1/tasks/leases`, and `/api/v1/tasks/salvage` as `post`-only `x-autogenerated` placeholders and omits explicit action/detail paths such as `/api/v1/tasks/queue/{task_id}/claim`, `/api/v1/tasks/queue/sync`, and `/api/v1/tasks/leases/{lease_id}/complete`. Strict SDK parity reports `TaskQueueHandler` with `5` routes and `0` Python / `0` TypeScript SDK coverage, while still passing only because the baseline budget permits the three task routes. `pytest -q tests/handlers/tasks/test_queue.py` passes (`7 passed`), proving runtime behavior exists while public contract coverage is stale. Open PRs are empty; open `boss-ready` issues are empty; existing issue `#5953` contains this handoff but is unlabeled, and adding `boss-ready` failed due GitHub API connectivity.
Acceptance Criteria: Add explicit OpenAPI definitions for the live TaskQueueHandler contract instead of relying on autogenerated placeholders; base `/api/v1/tasks/queue`, `/api/v1/tasks/leases`, and `/api/v1/tasks/salvage` must expose GET reads, not bogus base POST placeholders. Document supported action/detail subroutes: queue stats, task detail, queue sync, task claim, lease heartbeat, lease release, and lease completion. Add or extend focused OpenAPI tests proving generated schema methods match `TaskQueueHandler.handle()` / `handle_post()` dispatch and that no autogenerated placeholder remains for the task queue base paths. Extend Python and TypeScript SDK task namespaces with clearly named queue/lease/salvage methods without breaking existing `/api/v2/tasks` task-management methods. Add focused SDK route-mapping tests and keep scope bounded to TaskQueueHandler contract coverage.
Validation: Run `pytest -q tests/handlers/tasks/test_queue.py`. Run the focused OpenAPI test file for TaskQueueHandler routes. Run `python3 scripts/export_openapi.py --output-dir /tmp/founder-triage-task-queue-contract-openapi` and a direct JSON check that `/api/v1/tasks/queue`, `/api/v1/tasks/leases`, and `/api/v1/tasks/salvage` expose GET and are not `x-autogenerated` placeholders, plus checks for the supported action subpaths. Run `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json`. Run `python3 -m pytest -q sdk/python/tests/test_tasks.py` or the new focused Python SDK test. Run the focused TypeScript task namespace test or `npm --prefix sdk/typescript run typecheck` when dependencies are available.
Expiration Hours: 48
Backup Task: If issue #5953 is already actively fixed before pickup, take `origin/codex/codex-20260416-124216-0a47a9dc` (`fix(openapi): publish agent evolution action methods`) through validation and PR publication if it is still unmerged; otherwise cover `PromptEngineHandler` SDK parity next because `check_sdk_parity.py` reports eight prompt-engine routes missing from both SDKs and those routes support the prompt-to-spec/debate/execution wedge.

## 2026-04-16T11:54:01Z

- Runtime: about `1:03:44` since the last founder-triage run timestamp.
- Read `ROADMAP.md`, `docs/FEATURE_DISCOVERY.md`, canonical `docs/status/FEATURE_DISCOVERY.md`, `docs/status/NEXT_STEPS_CANONICAL.md`, `docs/status/ACTIVE_EXECUTION_ISSUES.md`, current founder-triage memory, and recent `founder-review`, `engineering-automation-2`, and `aragora-overnight-steward` memories.
- Current `origin/main` / `HEAD` is `cc413313e` (`fix(storage): serialize sqlite blacklist initialization (#5950)`). Recent merged PRs reviewed include `#5946`, `#5947`, `#5948`, and `#5950`; the prior SQLite write and initialization handoffs are now on main.
- Live duplication checks: `gh pr list --state open` returned `[]`; `gh issue list --label boss-ready` returned `[]`; targeted open PR searches for decision-analytics/OpenAPI/version-alignment returned `[]`. Targeted issue search did not find this exact generated-artifact drift; it only surfaced stale issue `#5929` for ChatHandler webhooks. Remote branch scan found unrelated fresh `origin/codex/codex-20260416-114159-58792897` (`fix(sdk): cover system intelligence routes`) and no branch fixing this decision-analytics OpenAPI generator drift.
- Proof surfaces remain fresh enough: `docs/status/B0_BENCHMARK_TRUTH_STATUS.md` last updated `2026-04-15T23:05:41Z`, coverage `5/5`, truth/no-rescue/merged rates `100%`; `docs/status/TW03_RESCUE_PRODUCTIZATION_STATUS.md` last updated `2026-04-15T23:05:42Z`, repeated rescue classes `0`.
- Local focused validation of the just-landed SQLite fixes passed: `python3 -m pytest -q tests/storage/test_token_blacklist_store.py tests/test_token_blacklist_store.py tests/test_storage_token_blacklist.py tests/test_token_revocation_race.py` => `118 passed, 2 skipped`.
- Current route validator state on main remains `96.0%` coverage with `91` missing handler routes and `39` orphaned spec routes. Current strict SDK parity passes within budget with `29` missing-from-both routes, `0` baseline regressions, and budget `29/29`; `python3 scripts/generate_sdk_types.py --openapi docs/api/openapi_generated.json --check` also passes.
- New proof-gate regression evidence: `gh pr view 5950 --json statusCheckRollup` shows `Version Alignment` failed in run `24507919787`, job `71631133296`, even though `#5950` merged. `gh run view 24507919787 --job 71631133296 --log-failed` shows the exact `Check OpenAPI spec is up to date` step failed while diffing regenerated OpenAPI against `docs/api/openapi_generated.json`.
- Local reproduction of the failed Version Alignment generation sequence on current main shows `scripts/generate_openapi.py` produces deprecated unversioned `/api/decision-analytics/{overview,trends,outcomes,agents,domains}` placeholders and downgrades the `/api/v1/decision-analytics/*` operations to autogenerated placeholders, while checked-in `docs/api/openapi_generated.json` contains rich versioned decision-analytics entries and no unversioned aliases.
- Root-cause anchor: `DecisionAnalyticsHandler.ROUTES` contains both versioned and legacy unversioned paths, while its `@api_endpoint` decorators define rich metadata only for the versioned paths. The multi-tier generator keeps canonical placeholder routes before the richer decorator metadata can override them.
- Aragora usage: `aragora spec` was run for this candidate and degraded to fallback-only output because the `spec-agent` circuit breaker is open.
- Tried to open a GitHub issue with title `Fix decision analytics OpenAPI generation drift` and label `boss-ready`; `gh issue create --repo synaptent/aragora ... --body-file -` failed with exact blocker: `error connecting to api.github.com` then `check your internet connection or https://githubstatus.com`.
- Publisher retry should create one `boss-ready` issue using the handoff below; do not create duplicates if a later run or publisher has already opened it.
- Current run time: `2026-04-16T11:54:01Z`.

Handoff Source: Founder triage automation on 2026-04-16; reviewed origin/main@cc413313e, roadmap/status docs, recent founder-review/engineering-automation-2/overnight-steward memories, live PR/issues, recent merged PRs #5946/#5948/#5950, current OpenAPI/SDK validators, and duplicate remote branches.
Priority: HIGH
Task Title: Fix decision analytics OpenAPI generation drift
Why Now: Current main merged the decision-analytics live route export and SDK coverage, but `Tests / Version Alignment` is still red on the latest merged PR because `scripts/generate_openapi.py` no longer reproduces the checked-in decision-analytics generated artifacts. This is an integration-correctness and pipeline-reliability gate: future PRs can inherit a red generated OpenAPI check, and the live `/api/v1/decision-analytics/*` contract can regress back to placeholder docs or duplicate legacy aliases.
Repo Evidence: `gh pr view 5950 --json statusCheckRollup` shows `Version Alignment` failed in run `24507919787`, job `71631133296`, even though #5950 merged. `gh run view 24507919787 --job 71631133296 --log-failed` shows the exact `Check OpenAPI spec is up to date` step failed while diffing a regenerated temp spec against `docs/api/openapi_generated.json`. Local reproduction with the same generation sequence shows the temp generator adds deprecated unversioned `/api/decision-analytics/{overview,trends,outcomes,agents,domains}` placeholders and downgrades `/api/v1/decision-analytics/{overview,trends,outcomes,agents,domains}` to autogenerated placeholder operations, while the checked-in docs keep only rich versioned entries. `DecisionAnalyticsHandler.ROUTES` contains both versioned and legacy unversioned paths, while its `@api_endpoint` decorators define rich metadata only for versioned paths. `python3 scripts/generate_sdk_types.py --openapi docs/api/openapi_generated.json --check` passes, so the immediate red gate is OpenAPI generation drift, not SDK type drift. `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json` passes within budget with `0` baseline regressions.
Acceptance Criteria: Make the exact Version Alignment OpenAPI generation sequence reproduce checked-in generated artifacts on current main. Preserve rich `/api/v1/decision-analytics/*` operation metadata and do not reintroduce deprecated unversioned `/api/decision-analytics/*` placeholders unless the product intentionally supports and documents those aliases. Add focused regression coverage so `DecisionAnalyticsHandler` OpenAPI generation prefers decorator metadata over legacy `ROUTES` placeholders for duplicate versioned paths and either suppresses or explicitly documents legacy aliases consistently. Keep scope to decision-analytics/OpenAPI generation drift; do not broaden into the full remaining 91 missing routes or generic SDK parity debt.
Validation: Run the exact failed sequence locally: generate temp JSON/YAML with `scripts/generate_openapi.py`, run `scripts/add_openapi_operation_ids.py`, `scripts/add_openapi_param_descriptions.py`, and `scripts/add_openapi_descriptions.py`, then `diff -u` both temp files against `docs/api/openapi_generated.json` and `.yaml`. Run `python3 scripts/validate_openapi_routes.py --spec docs/api/openapi.json --json`. Run `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json`. Run focused tests around `tests/server/openapi/test_live_dashboard_route_exports.py` and any new generator regression test.
Expiration Hours: 24
Backup Task: If this OpenAPI generation drift is already fixed before pickup, publish or recreate the existing remote SDK coverage branch `origin/codex/codex-20260416-114159-58792897` (`fix(sdk): cover system intelligence routes`) after reconciling with current main; it touches only `sdk/python/aragora_sdk/namespaces/system.py`, `sdk/python/tests/test_system.py`, `sdk/typescript/src/namespaces/system.ts`, and its TypeScript tests.

## 2026-04-16T10:53:08Z

- Runtime: about `1:02:51` since the last founder-triage memory entry.
- Read `ROADMAP.md`, `docs/FEATURE_DISCOVERY.md`, canonical `docs/status/FEATURE_DISCOVERY.md`, `docs/status/NEXT_STEPS_CANONICAL.md`, `docs/status/ACTIVE_EXECUTION_ISSUES.md`, current founder-triage memory, and recent `founder-review`, `engineering-automation-2`, and `aragora-overnight-steward` memories.
- Current `origin/main` / `HEAD` is `4bed075d6` (`fix(openapi): restore decision analytics routes (#5946)`). Recent merged PRs include `#5942`, `#5943`, `#5944`, `#5945`, and `#5946`; the prior decision-analytics OpenAPI and SDK parity handoff is now complete on main.
- Live GitHub duplication checks: `gh pr list --state open` returned `[]`; `gh issue list --label boss-ready` returned `[]`; targeted issue/PR searches for `token_blacklist`, `SQLiteBlacklist`, `database is locked`, and SDK coverage found no open duplicate for the primary. Open issues `#5929` and `#5935` remain open but appear stale after the relevant route/export fixes landed; they are not the next coding task.
- Proof surfaces remain fresh enough: `docs/status/B0_BENCHMARK_TRUTH_STATUS.md` last updated `2026-04-15T23:05:41Z`, coverage `5/5`, truth/no-rescue/merged rates `100%`; `docs/status/TW03_RESCUE_PRODUCTIZATION_STATUS.md` last updated `2026-04-15T23:05:42Z`, repeated rescue classes `0`.
- Current strict SDK parity on main passes within budget: `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json` reports `29` missing-from-both routes, `0` baseline regressions, and budget `29/29`.
- Current route validator state on main: `python3 scripts/validate_openapi_routes.py --spec docs/api/openapi.json --json` reports `96.0%` coverage, `91` missing handler routes, and `39` orphaned spec routes.
- Post-merge CI evidence for the next proof-first/pipeline task: `gh run view 24501689313 --job 71611666182 --log-failed` shows PR `#5944`'s storage shard failed at `tests/storage/test_token_blacklist_store.py::TestSQLiteBlacklist::test_concurrent_access` with `AssertionError: assert 1 == 0`, where the only error was `OperationalError('database is locked')`; shard summary was `1 failed, 6028 passed, 19 skipped`.
- Local focused repro check on current main did not fail once: `python3 -m pytest -q tests/storage/test_token_blacklist_store.py::TestSQLiteBlacklist::test_concurrent_access` passed. This supports a concurrency hardening task rather than a deterministic local regression.
- Duplication branch check found exact unpublished remote work: `origin/codex/codex-20260416-104216-9dc3b80f` contains commit `0933f1ae1 fix(storage): serialize sqlite blacklist writes`, touching only `aragora/storage/token_blacklist_store.py` and `tests/storage/test_token_blacklist_store.py`; no open PR exists for that head branch.
- The branch patch adds a SQLite busy timeout, serializes writes/cleanup with an `RLock`, and adds `test_configures_busy_timeout`. Because the exact fix already exists on a remote codex branch, no new GitHub issue was opened.
- Aragora usage: `aragora spec` was run to compare the SQLite pipeline-reliability task against baseline SDK coverage expansion; it degraded to fallback-only output because the `spec-agent` circuit breaker is open.
- Decision: primary recommendation is to finish/publish the existing SQLite token blacklist concurrency branch, because it addresses a concrete post-merge proof-gate failure on a shared storage path and is smaller/more urgent than baseline-tolerated SDK coverage breadth.
- Current run time: `2026-04-16T10:53:08Z`.

Handoff Source: Founder triage automation on 2026-04-16; reviewed `origin/main@4bed075d6`, roadmap/status docs, recent automation memories, live PR/issues, recent merged PRs, current SDK/route validator state, post-merge CI logs for `#5944`, and duplicate remote branches.
Priority: HIGH
Task Title: Publish SQLite blacklist concurrency fix
Why Now: Main carries a post-merge CI storage-shard failure from `#5944`: `TestSQLiteBlacklist.test_concurrent_access` hit `OperationalError('database is locked')` under GitHub's parallel storage/ranking/persistence/billing shard. This is a proof-gate/pipeline reliability issue on an auth-token storage path, and a focused remote branch already contains the likely bounded fix but has no open PR.
Repo Evidence: `gh run view 24501689313 --job 71611666182 --log-failed` shows `tests/storage/test_token_blacklist_store.py::TestSQLiteBlacklist::test_concurrent_access` failed with `AssertionError: assert 1 == 0`, where `errors == [OperationalError('database is locked')]`, and the shard summary was `1 failed, 6028 passed, 19 skipped`. Current main's focused local test passed once, so treat this as concurrency/CI hardening rather than deterministic local failure. `origin/codex/codex-20260416-104216-9dc3b80f` is ahead of main by `0933f1ae1 fix(storage): serialize sqlite blacklist writes`, touching only `aragora/storage/token_blacklist_store.py` and `tests/storage/test_token_blacklist_store.py`; no open PR or boss-ready issue duplicates it.
Acceptance Criteria: Port or reuse `origin/codex/codex-20260416-104216-9dc3b80f` on current main, preserving the narrow scope: configure SQLite busy timeout for `SQLiteBlacklist`, serialize writes and cleanup without blocking reads more than necessary, add/keep focused coverage for busy-timeout configuration and concurrent access, and avoid broad storage refactors. Open a PR from the branch or an equivalent fresh branch after validation.
Validation: Run `python3 -m pytest -q tests/storage/test_token_blacklist_store.py`; run a repeated stress loop for `tests/storage/test_token_blacklist_store.py::TestSQLiteBlacklist::test_concurrent_access` to increase confidence; run the affected CI shard command or closest local equivalent for `tests/storage tests/ranking tests/persistence tests/billing` if runtime allows; confirm `gh pr checks` is green after publishing.
Expiration Hours: 24
Backup Task: If the SQLite concurrency branch has already landed, add SDK coverage for the highest-value baseline-missing live route families still reported by `scripts/check_sdk_parity.py`: start with `SystemIntelligenceHandler` and `AgentEvolutionDashboardHandler`, then `FeedbackHubHandler` or `ObservabilityDashboardHandler`, keeping the change to one SDK namespace slice with focused Python and TypeScript tests.

## 2026-04-16T09:53:33Z

- Runtime: about `1:03:16` since the last founder-triage run timestamp.
- Read `ROADMAP.md`, `docs/FEATURE_DISCOVERY.md`, canonical `docs/status/FEATURE_DISCOVERY.md`, `docs/status/NEXT_STEPS_CANONICAL.md`, `docs/status/ACTIVE_EXECUTION_ISSUES.md`, current founder-triage memory, and recent `founder-review`, `engineering-automation-2`, and `aragora-overnight-steward` memories.
- Current local `origin/main` / `HEAD` is `0165410f9` (`fix(openapi): document inbox trust wedge receipts (#5945)`). Recent merged PRs include `#5942`, `#5943`, `#5944`, and `#5945`; the previous inbox trust-wedge OpenAPI gap is now on main.
- Proof surfaces remain fresh enough for this run: `docs/status/B0_BENCHMARK_TRUTH_STATUS.md` last updated `2026-04-15T23:05:41Z`, coverage `5/5`, truth/no-rescue/merged rates `100%`; `docs/status/TW03_RESCUE_PRODUCTIZATION_STATUS.md` last updated `2026-04-15T23:05:42Z`, repeated rescue classes `0`.
- Live duplication checks: `gh pr list --state open` returned exactly one open PR, draft `#5946` (`fix(openapi): restore decision analytics routes`) on `codex/codex-20260416-094230-22181582`; open `boss-ready` issue list returned `[]`. Targeted issue search found existing issue `#5890` for the same decision-analytics OpenAPI export gap and no `token_blacklist_store` / SQLite lock issue.
- `#5946` changes only `aragora/server/handlers/_lazy_imports.py` and `tests/server/openapi/test_live_dashboard_route_exports.py`; it fixes existing issue `#5890` by adding `DecisionAnalyticsHandler` to OpenAPI discovery and focused tests.
- Direct `#5946` CI evidence: required checks `Generate & Validate`, `TypeScript SDK Type Check`, `lint`, and `typecheck` pass; `sdk-parity` fails. The failed `sdk-parity-run` reports five baseline regressions: `/api/decision-analytics/agents`, `/domains`, `/outcomes`, `/overview`, and `/trends` are newly missing from both SDKs after the handler enters `ALL_HANDLERS`.
- Local baseline check on current main passes before `#5946`: `python3 scripts/check_sdk_parity.py --strict --baseline scripts/baselines/check_sdk_parity.json --budget scripts/baselines/check_sdk_parity_budget.json` reports `29` missing-from-both routes, `0` baseline regressions, and budget `29/29`.
- Current docs/API route validator state on main: `python3 scripts/validate_openapi_routes.py --spec docs/api/openapi.json --json` reports `96.0%` coverage, `91` missing handler routes, and `39` orphaned spec routes.
- Backup investigation: overnight steward observed a post-merge `#5944` storage shard failure in `tests/storage/test_token_blacklist_store.py::TestSQLiteBlacklist::test_concurrent_access` with one `OperationalError('database is locked')`; a focused local rerun of that single test passed (`1 passed in 0.49s`), so this is backup CI stability work rather than the primary live-path unblock.
- Aragora usage:

---
Automation publisher truncated this issue body because it exceeded GitHub's size limit. See the source automation memory path above for full evidence.

--- a/aragora/swarm/github_app_auth.py
+++ b/aragora/swarm/github_app_auth.py
@@ -38,15 +38,44 @@ def _read_env_file(path: Path) -> dict[str, str]:
     if not path.exists():
         return {}
     values: dict[str, str] = {}
-    for raw_line in path.read_text(encoding="utf-8").splitlines():
+    lines = path.read_text(encoding="utf-8").splitlines()
+    index = 0
+    while index < len(lines):
+        raw_line = lines[index]
         line = raw_line.strip()
         if not line or line.startswith("#") or "=" not in line:
+            index += 1
             continue
         key, value = line.split("=", 1)
         key = key.strip()
-        value = value.strip().strip("'\"")
-        if key:
-            values[key] = value
+        value = value.strip()
+        if not key:
+            index += 1
+            continue
+
+        if value.startswith(("'", '"')):
+            quote = value[0]
+            value = value[1:]
+            if value.endswith(quote):
+                values[key] = value[:-1]
+                index += 1
+                continue
+
+            parts = [value]
+            index += 1
+            while index < len(lines):
+                next_line = lines[index]
+                if next_line.endswith(quote):
+                    parts.append(next_line[:-1])
+                    index += 1
+                    break
+                parts.append(next_line)
+                index += 1
+            values[key] = "\n".join(parts)
+            continue
+
+        values[key] = value
+        index += 1
     return values
 
 
@@ -70,6 +99,12 @@ def _automation_env_file(env: Mapping[str, str]) -> Path:
     return Path(configured).expanduser() if configured else DEFAULT_AUTOMATION_ENV_FILE
 
 
+def _normalize_private_key(value: str) -> str:
+    if "\\n" in value:
+        return value.replace("\\n", "\n")
+    return value
+
+
 def load_github_app_config(env: Mapping[str, str] | None = None) -> GitHubAppConfig | None:
     base_env = dict(os.environ if env is None else env)
     file_env = _read_env_file(_automation_env_file(base_env))
@@ -86,6 +121,8 @@ def load_github_app_config(env: Mapping[str, str] | None = None) -> GitHubAppCon
     )
     private_key = _first_value(values, ("GITHUB_APP_PRIVATE_KEY", "ARAGORA_GITHUB_APP_KEY"))
     private_key_source = "env:GITHUB_APP_PRIVATE_KEY" if private_key else ""
+    if private_key:
+        private_key = _normalize_private_key(private_key)
     if not private_key:
         key_path = _first_value(
             values,

--- a/scripts/publish_automation_handoffs.py
+++ b/scripts/publish_automation_handoffs.py
@@ -26,6 +26,7 @@ DEFAULT_LABELS = ("boss-ready",)
 DEFAULT_LIMIT = 2
 DEFAULT_MAX_OPEN_ISSUES = 12
 DEFAULT_COMMAND_TIMEOUT_SECONDS = 45
+MAX_ISSUE_BODY_CHARS = 60_000
 DEFAULT_AUTOMATION_IDS = (
     "founder-review",
     "founder-triage",
@@ -433,7 +434,7 @@ def _create_issue(
         "--title",
         handoff.task_title,
         "--body",
-        handoff.body,
+        _fit_issue_body(handoff.body),
     ]
     for label in labels:
         args.extend(["--label", label])
@@ -443,6 +444,17 @@ def _create_issue(
     url = str(proc.stdout or "").strip().splitlines()[-1].strip()
     _add_issue_labels(repo_root, repo, url, labels)
     return url
+
+
+def _fit_issue_body(body: str) -> str:
+    if len(body) <= MAX_ISSUE_BODY_CHARS:
+        return body
+    suffix = (
+        "\n\n---\n"
+        "Automation publisher truncated this issue body because it exceeded "
+        "GitHub's size limit. See the source automation memory path above for full evidence."
+    )
+    return body[: MAX_ISSUE_BODY_CHARS - len(suffix)].rstrip() + suffix
 
 
 def _issue_number_from_url(url: str) -> str | None:

--- a/scripts/publish_automation_handoffs.py
+++ b/scripts/publish_automation_handoffs.py
@@ -25,6 +25,7 @@ DEFAULT_REPO = "synaptent/aragora"
 DEFAULT_LABELS = ("boss-ready",)
 DEFAULT_LIMIT = 2
 DEFAULT_MAX_OPEN_ISSUES = 12
+DEFAULT_COMMAND_TIMEOUT_SECONDS = 45
 DEFAULT_AUTOMATION_IDS = (
     "founder-review",
     "founder-triage",
@@ -66,6 +67,17 @@ STOPWORDS = {
     "with",
 }
 
+try:
+    from aragora.swarm.github_app_auth import github_cli_env
+except Exception:  # pragma: no cover - fallback for partially bootstrapped script contexts
+
+    def github_cli_env(
+        base_env: dict[str, str] | None = None,
+        *,
+        prefer_app: bool = True,
+    ) -> dict[str, str]:
+        return dict(os.environ if base_env is None else base_env)
+
 
 @dataclass(frozen=True)
 class Handoff:
@@ -89,7 +101,25 @@ class PublishDecision:
 
 
 def _run(args: list[str], *, cwd: Path) -> subprocess.CompletedProcess[str]:
-    return subprocess.run(args, cwd=cwd, text=True, capture_output=True, check=False)
+    env = github_cli_env(os.environ) if args and args[0] == "gh" else None
+    try:
+        return subprocess.run(
+            args,
+            cwd=cwd,
+            text=True,
+            capture_output=True,
+            check=False,
+            timeout=DEFAULT_COMMAND_TIMEOUT_SECONDS,
+            env=env,
+        )
+    except subprocess.TimeoutExpired as exc:
+        stdout = exc.stdout if isinstance(exc.stdout, str) else ""
+        stderr = exc.stderr if isinstance(exc.stderr, str) else ""
+        message = (
+            stderr
+            or f"command timed out after {DEFAULT_COMMAND_TIMEOUT_SECONDS}s: {' '.join(args)}"
+        )
+        return subprocess.CompletedProcess(args=args, returncode=124, stdout=stdout, stderr=message)
 
 
 def _codex_home(value: str | None) -> Path:
@@ -410,7 +440,28 @@ def _create_issue(
     proc = _run(args, cwd=repo_root)
     if proc.returncode != 0:
         raise RuntimeError(proc.stderr.strip() or proc.stdout.strip() or "gh issue create failed")
-    return str(proc.stdout or "").strip().splitlines()[-1].strip()
+    url = str(proc.stdout or "").strip().splitlines()[-1].strip()
+    _add_issue_labels(repo_root, repo, url, labels)
+    return url
+
+
+def _issue_number_from_url(url: str) -> str | None:
+    match = re.search(r"/issues/(\d+)(?:$|[/?#])", url)
+    return match.group(1) if match else None
+
+
+def _add_issue_labels(repo_root: Path, repo: str, issue_url: str, labels: list[str]) -> None:
+    if not labels:
+        return
+    number = _issue_number_from_url(issue_url)
+    if not number:
+        return
+    proc = _run(
+        ["gh", "issue", "edit", number, "--repo", repo, "--add-label", ",".join(labels)],
+        cwd=repo_root,
+    )
+    if proc.returncode != 0:
+        return
 
 
 def decide_handoffs(

--- a/scripts/publish_codex_automation_branches.py
+++ b/scripts/publish_codex_automation_branches.py
@@ -11,6 +11,7 @@ from __future__ import annotations
 
 import argparse
 import json
+import os
 import subprocess
 import sys
 from dataclasses import asdict, dataclass
@@ -22,6 +23,9 @@ UTC = timezone.utc
 DEFAULT_SINCE_HOURS = 72
 DEFAULT_PUBLISH_LIMIT = 1
 DEFAULT_MAX_OPEN_PRS = 1
+DEFAULT_COMMAND_TIMEOUT_SECONDS = 45
+DEFAULT_GIT_TIMEOUT_SECONDS = 15
+DEFAULT_SCAN_LIMIT = 12
 CODEX_BRANCH_PREFIX = "codex/"
 DEFAULT_PREFLIGHT_SCRIPT = "scripts/automation_pr_preflight.sh"
 ACTIVE_SESSION_FILES = (
@@ -29,6 +33,17 @@ ACTIVE_SESSION_FILES = (
     ".codex_session_active",
     ".nomic-session-active",
 )
+
+try:
+    from aragora.swarm.github_app_auth import github_cli_env
+except Exception:  # pragma: no cover - fallback for partially bootstrapped script contexts
+
+    def github_cli_env(
+        base_env: dict[str, str] | None = None,
+        *,
+        prefer_app: bool = True,
+    ) -> dict[str, str]:
+        return dict(os.environ if base_env is None else base_env)
 
 
 @dataclass(frozen=True)
@@ -69,13 +84,25 @@ def _run(
     cwd: Path,
     check: bool = False,
 ) -> subprocess.CompletedProcess[str]:
-    return subprocess.run(
-        args,
-        cwd=cwd,
-        text=True,
-        capture_output=True,
-        check=check,
+    env = github_cli_env(os.environ) if args and args[0] == "gh" else None
+    timeout = (
+        DEFAULT_COMMAND_TIMEOUT_SECONDS if args and args[0] == "gh" else DEFAULT_GIT_TIMEOUT_SECONDS
     )
+    try:
+        return subprocess.run(
+            args,
+            cwd=cwd,
+            text=True,
+            capture_output=True,
+            check=check,
+            timeout=timeout,
+            env=env,
+        )
+    except subprocess.TimeoutExpired as exc:
+        stdout = exc.stdout if isinstance(exc.stdout, str) else ""
+        stderr = exc.stderr if isinstance(exc.stderr, str) else ""
+        message = stderr or f"command timed out after {timeout}s: {' '.join(args)}"
+        return subprocess.CompletedProcess(args=args, returncode=124, stdout=stdout, stderr=message)
 
 
 def _parse_git_dt(value: str) -> datetime:
@@ -109,6 +136,14 @@ def _branch_unique_commit_count(repo_root: Path, base: str, branch: str) -> int:
 def _branch_is_merged(repo_root: Path, base: str, branch: str) -> bool:
     proc = _run(["git", "merge-base", "--is-ancestor", branch, base], cwd=repo_root)
     return proc.returncode == 0
+
+
+def _branch_patch_equivalent_to_base(repo_root: Path, base: str, branch: str) -> bool:
+    proc = _run(["git", "cherry", base, branch], cwd=repo_root)
+    if proc.returncode != 0:
+        return False
+    statuses = [line[:1] for line in proc.stdout.splitlines() if line.strip()]
+    return bool(statuses) and all(status == "-" for status in statuses)
 
 
 def _local_codex_branches(repo_root: Path) -> list[BranchSnapshot]:
@@ -155,7 +190,11 @@ def _worktree_is_dirty(path: Path) -> bool:
     return bool(proc.stdout.strip())
 
 
-def _list_worktrees(repo_root: Path) -> list[WorktreeSnapshot]:
+def _list_worktrees(
+    repo_root: Path,
+    *,
+    branch_filter: set[str] | None = None,
+) -> list[WorktreeSnapshot]:
     proc = _run(["git", "worktree", "list", "--porcelain"], cwd=repo_root)
     if proc.returncode != 0:
         raise RuntimeError(proc.stderr.strip() or "failed to list worktrees")
@@ -167,6 +206,8 @@ def _list_worktrees(repo_root: Path) -> list[WorktreeSnapshot]:
 
     def flush() -> None:
         if current_path is None:
+            return
+        if branch_filter is not None and current_branch not in branch_filter:
             return
         snapshots.append(
             WorktreeSnapshot(
@@ -264,6 +305,7 @@ def select_publishable_branches(
     cutoff: datetime,
     base: str,
     is_merged: dict[str, bool] | None = None,
+    is_patch_equivalent: dict[str, bool] | None = None,
     historical_pr_branches: set[str] | None = None,
 ) -> list[PublishDecision]:
     worktrees_by_branch: dict[str, list[WorktreeSnapshot]] = {}
@@ -272,6 +314,7 @@ def select_publishable_branches(
             worktrees_by_branch.setdefault(worktree.branch, []).append(worktree)
 
     merged_lookup = is_merged or {}
+    patch_equivalent_lookup = is_patch_equivalent or {}
     historical_lookup = historical_pr_branches or set()
     decisions: list[PublishDecision] = []
 
@@ -281,6 +324,8 @@ def select_publishable_branches(
 
         if merged_lookup.get(branch.branch, False):
             reason = "already_merged"
+        elif patch_equivalent_lookup.get(branch.branch, False):
+            reason = "patch_equivalent_to_base"
         elif branch.unique_commit_count <= 0:
             reason = "no_unique_commits"
         elif branch.committed_at < cutoff:
@@ -512,6 +557,12 @@ def _build_parser() -> argparse.ArgumentParser:
         help="Restrict evaluation to one or more explicit branch names",
     )
     parser.add_argument(
+        "--scan-limit",
+        type=int,
+        default=DEFAULT_SCAN_LIMIT,
+        help="Maximum recent local codex branches to inspect when --branch is not provided.",
+    )
+    parser.add_argument(
         "--label",
         action="append",
         dest="labels",
@@ -555,10 +606,22 @@ def main(argv: list[str] | None = None) -> int:
     if args.branches:
         requested = set(args.branches)
         branches = [branch for branch in branches if branch.branch in requested]
+    else:
+        branches = sorted(
+            [branch for branch in branches if branch.committed_at >= cutoff],
+            key=lambda branch: branch.committed_at,
+            reverse=True,
+        )[: max(args.scan_limit, 0)]
 
-    worktrees = _list_worktrees(repo_root)
+    branch_names = {branch.branch for branch in branches}
+    worktrees = _list_worktrees(repo_root, branch_filter=branch_names)
     merged_lookup = {
         branch.branch: _branch_is_merged(repo_root, args.base, branch.branch) for branch in branches
+    }
+    patch_equivalent_lookup = {
+        branch.branch: _branch_patch_equivalent_to_base(repo_root, args.base, branch.branch)
+        for branch in branches
+        if not merged_lookup.get(branch.branch, False)
     }
     hydrated_branches = [
         BranchSnapshot(
@@ -585,6 +648,7 @@ def main(argv: list[str] | None = None) -> int:
         cutoff=cutoff,
         base=args.base,
         is_merged=merged_lookup,
+        is_patch_equivalent=patch_equivalent_lookup,
         historical_pr_branches=historical_pr_branches,
     )
 

--- a/scripts/run_codex_automation_publisher.sh
+++ b/scripts/run_codex_automation_publisher.sh
@@ -25,15 +25,27 @@ if ! command -v gh >/dev/null 2>&1; then
   exit 1
 fi
 
-if ! gh auth status >/dev/null 2>&1; then
-  echo "$(STAMP) [codex-automation-publisher] gh auth unavailable; skipping publish pass"
-  exit 0
+if ! git fetch --no-write-fetch-head --prune origin '+refs/heads/*:refs/remotes/origin/*' >/dev/null 2>&1; then
+  echo "$(STAMP) [codex-automation-publisher] origin refresh failed; continuing with cached refs"
 fi
 
 echo "$(STAMP) [codex-automation-publisher] starting handoff publish pass"
-python3 scripts/publish_automation_handoffs.py --apply --limit 1 --max-open-issues 12 --json
-echo "$(STAMP) [codex-automation-publisher] handoff publish pass complete"
+if python3 scripts/publish_automation_handoffs.py --apply --limit 1 --max-open-issues 12 --json; then
+  echo "$(STAMP) [codex-automation-publisher] handoff publish pass complete"
+else
+  echo "$(STAMP) [codex-automation-publisher] handoff publish pass failed; continuing"
+fi
 
 echo "$(STAMP) [codex-automation-publisher] starting branch publish pass"
-python3 scripts/publish_codex_automation_branches.py --apply --limit 1 --max-open-prs 1 --json
+if python3 scripts/publish_codex_automation_branches.py \
+  --base origin/main \
+  --apply \
+  --limit 1 \
+  --max-open-prs 1 \
+  --json; then
+  echo "$(STAMP) [codex-automation-publisher] branch publish pass complete"
+else
+  echo "$(STAMP) [codex-automation-publisher] branch publish pass failed"
+fi
+
 echo "$(STAMP) [codex-automation-publisher] publish pass complete"

--- a/tests/scripts/test_publish_automation_handoffs.py
+++ b/tests/scripts/test_publish_automation_handoffs.py
@@ -292,3 +292,34 @@ def test_publish_handoffs_creates_issue_with_labels(monkeypatch: Any, tmp_path: 
         "--add-label",
         "boss-ready,autonomous",
     ]
+
+
+def test_create_issue_truncates_oversized_body(monkeypatch: Any, tmp_path: Path) -> None:
+    bodies: list[str] = []
+
+    def fake_run(args: list[str], cwd: Path) -> subprocess.CompletedProcess[str]:
+        if args[:3] == ["gh", "issue", "create"]:
+            bodies.append(args[args.index("--body") + 1])
+            return subprocess.CompletedProcess(
+                args, 0, "https://github.com/synaptent/aragora/issues/6000\n", ""
+            )
+        return subprocess.CompletedProcess(args, 0, "", "")
+
+    monkeypatch.setattr(mod, "_run", fake_run)
+
+    mod._create_issue(
+        tmp_path,
+        "synaptent/aragora",
+        Handoff(
+            source_file=str(tmp_path / "memory.md"),
+            task_title="Long issue body",
+            priority="HIGH",
+            body="x" * (mod.MAX_ISSUE_BODY_CHARS + 1000),
+            labels={},
+            expires_at=None,
+        ),
+        labels=["boss-ready"],
+    )
+
+    assert len(bodies[0]) <= mod.MAX_ISSUE_BODY_CHARS
+    assert "truncated this issue body" in bodies[0]

--- a/tests/scripts/test_publish_automation_handoffs.py
+++ b/tests/scripts/test_publish_automation_handoffs.py
@@ -282,3 +282,13 @@ def test_publish_handoffs_creates_issue_with_labels(monkeypatch: Any, tmp_path: 
     assert published[0].created_issue_url == "https://github.com/synaptent/aragora/issues/5890"
     assert created[0][:3] == ["gh", "issue", "create"]
     assert created[0].count("--label") == 2
+    assert created[1] == [
+        "gh",
+        "issue",
+        "edit",
+        "5890",
+        "--repo",
+        "synaptent/aragora",
+        "--add-label",
+        "boss-ready,autonomous",
+    ]

--- a/tests/scripts/test_publish_codex_automation_branches.py
+++ b/tests/scripts/test_publish_codex_automation_branches.py
@@ -70,6 +70,7 @@ def test_parser_defaults_to_single_branch_publish_budget() -> None:
 
     assert args.limit == 1
     assert args.max_open_prs == 1
+    assert args.scan_limit == mod.DEFAULT_SCAN_LIMIT
     assert args.skip_preflight is False
     assert args.preflight_script == mod.DEFAULT_PREFLIGHT_SCRIPT
 
@@ -80,6 +81,7 @@ def test_select_publishable_branches_skips_open_pr_and_old_or_merged_branches() 
             _branch("codex/already-open"),
             _branch("codex/old", hours_ago=200),
             _branch("codex/merged"),
+            _branch("codex/cherry-picked"),
             _branch("codex/no-unique", unique_commit_count=0),
         ],
         [],
@@ -90,8 +92,10 @@ def test_select_publishable_branches_skips_open_pr_and_old_or_merged_branches() 
             "codex/already-open": False,
             "codex/old": False,
             "codex/merged": True,
+            "codex/cherry-picked": False,
             "codex/no-unique": False,
         },
+        is_patch_equivalent={"codex/cherry-picked": True},
         historical_pr_branches=set(),
     )
 
@@ -99,6 +103,7 @@ def test_select_publishable_branches_skips_open_pr_and_old_or_merged_branches() 
     assert by_branch["codex/already-open"].reason == "open_pr_exists"
     assert by_branch["codex/old"].reason == "older_than_cutoff"
     assert by_branch["codex/merged"].reason == "already_merged"
+    assert by_branch["codex/cherry-picked"].reason == "patch_equivalent_to_base"
     assert by_branch["codex/no-unique"].reason == "no_unique_commits"
 
 
@@ -200,6 +205,39 @@ def test_worktree_is_dirty_ignores_untracked_files(tmp_path: Path) -> None:
 
     tracked.write_text("changed\n", encoding="utf-8")
     assert _worktree_is_dirty(repo) is True
+
+
+def test_list_worktrees_filters_before_dirty_checks(monkeypatch: Any, tmp_path: Path) -> None:
+    payload = """
+worktree /tmp/codex-a
+HEAD abc123
+branch refs/heads/codex/a
+
+worktree /tmp/codex-b
+HEAD def456
+branch refs/heads/codex/b
+""".strip()
+    dirty_checked: list[str] = []
+
+    monkeypatch.setattr(
+        mod,
+        "_run",
+        lambda args, cwd, check=False: subprocess.CompletedProcess(
+            args=args, returncode=0, stdout=payload, stderr=""
+        ),
+    )
+    monkeypatch.setattr(mod, "_has_active_session", lambda path: False)
+
+    def fake_dirty(path: Path) -> bool:
+        dirty_checked.append(str(path))
+        return False
+
+    monkeypatch.setattr(mod, "_worktree_is_dirty", fake_dirty)
+
+    snapshots = mod._list_worktrees(tmp_path, branch_filter={"codex/b"})
+
+    assert [snapshot.branch for snapshot in snapshots] == ["codex/b"]
+    assert dirty_checked == [str(Path("/tmp/codex-b").resolve())]
 
 
 def test_publish_decisions_respects_open_pr_cap(monkeypatch: Any, tmp_path: Path) -> None:

--- a/tests/swarm/test_github_app_auth.py
+++ b/tests/swarm/test_github_app_auth.py
@@ -29,6 +29,45 @@ def test_load_github_app_config_reads_automation_env_file(tmp_path) -> None:
     assert config.private_key_source == str(key_path)
 
 
+def test_load_github_app_config_reads_multiline_private_key(tmp_path) -> None:
+    env_path = tmp_path / ".env.automation"
+    env_path.write_text(
+        "\n".join(
+            [
+                "GITHUB_APP_ID=123",
+                "GITHUB_APP_INSTALLATION_ID=456",
+                'GITHUB_APP_PRIVATE_KEY="-----BEGIN RSA PRIVATE KEY-----',
+                "key-body",
+                '-----END RSA PRIVATE KEY-----"',
+            ]
+        ),
+        encoding="utf-8",
+    )
+
+    config = mod.load_github_app_config({"ARAGORA_AUTOMATION_ENV_FILE": str(env_path)})
+
+    assert config is not None
+    assert config.private_key == (
+        "-----BEGIN RSA PRIVATE KEY-----\nkey-body\n-----END RSA PRIVATE KEY-----"
+    )
+
+
+def test_load_github_app_config_decodes_escaped_newline_private_key() -> None:
+    config = mod.load_github_app_config(
+        {
+            "ARAGORA_AUTOMATION_ENV_FILE": "/missing",
+            "GITHUB_APP_ID": "123",
+            "GITHUB_APP_INSTALLATION_ID": "456",
+            "GITHUB_APP_PRIVATE_KEY": "-----BEGIN RSA PRIVATE KEY-----\\nkey-body\\n-----END RSA PRIVATE KEY-----",
+        }
+    )
+
+    assert config is not None
+    assert config.private_key == (
+        "-----BEGIN RSA PRIVATE KEY-----\nkey-body\n-----END RSA PRIVATE KEY-----"
+    )
+
+
 def test_github_cli_env_prefers_installation_token(monkeypatch, tmp_path) -> None:
     mod.clear_github_app_token_cache()
     key_path = tmp_path / "app.pem"

